pkg/lsp/client_pool.go

```diff
@@ -0,0 +1,66 @@
+package lsp
+
+import "sync"
+
+// In memory store for clients. Applies mutex locking for concurrent access.
+type ClientPool struct {
+	clients map[ProjectId]map[LanguageId]Client
+	mu      sync.Mutex
+}
+
+func (c *ClientPool) Get(projectId ProjectId, languageId LanguageId) (Client, bool) {
```

Comment:
Maybe it's a good idea to create a client if not exists as part of get.

Reply:
I had the same idea originally but hesitated implementing it for 2 reasons: (1) I don't fully understand the potential implications on latency, and (2) I'm not a big fan of methods doing many things e.g. `getOrCreate()`.

With that being said, I'm open to reconsider it in future. After all, some text editors like Neovim start LSP servers lazily, i.e. only when you open a file with a particular extension.